### PR TITLE
Add support for `unique` groupby aggregation

### PR DESCRIPTION
**Pull request by shwina**
Adds support for `SeriesGroupBy.unique()`. Also adds support for `DataFrameGroupBy.unique()` but that's not tested, as Pandas doesn't support it (yet?).

Resolves https://github.com/rapidsai/cudf/issues/2973

### CODE REVIEW

File: python/cudf/cudf/tests/test_groupby.py [hunk 1]
```diff
@@ -1532,3 +1538,26 @@ def test_groupby_nonempty_no_keys(pdf):
         lambda: gdf.groupby([]),
         compare_error_message=False,
     )
+
+
+@pytest.mark.parametrize(
+    "by,data",
+    [
+        # ([], []),  # error?
```
Comment (shwina) [inline python/cudf/cudf/tests/test_groupby.py:1546]: libcudf raises here, but maybe we should early-return anyway. Looking into this.

Comment (shwina) [reply python/cudf/cudf/tests/test_groupby.py:1546]: Same issue as https://github.com/rapidsai/cudf/issues/7728. Likely should handle in a separate fix.

**Comment by ttnghia** [comment]
Wait, is this the same as https://github.com/rapidsai/cudf/pull/7664?

**Comment by kkraus14** [comment]
> Wait, is this the same as #7664? Or does it have any overlapping work with that PR?

This is different. That PR is enabling elementwise `unique`, this is just the expected Python API to map to the libcudf groupby `collect_set` aggregation.

File: python/cudf/cudf/tests/test_groupby.py (hunk 1 above)
Comment (kkraus14) [reply python/cudf/cudf/tests/test_groupby.py:1546]: Should we add a catch in the Python side temporarily to at least raise a nice `NotImplementedError`?

Comment (shwina) [reply python/cudf/cudf/tests/test_groupby.py:1546]: Fixed. Another instance of us catching the libcudf error string..

**Review by kkraus14 — APPROVED** [review]

**Comment by kkraus14** [comment]
@gpucibot merge

**Comment by ttnghia** [comment]
Rerun tests.

**Pull request merged**